sys/net/gnrc/netif/gnrc_netif.c

```diff
@@ -52,6 +52,10 @@
 #define ENABLE_DEBUG 0
 #include "debug.h"
 
+#ifndef CONFIG_NETIF_NUM_MAX
+#define CONFIG_NETIF_NUM_MAX 2
+#endif
```

Comment:
Unrelated?

Reply:
Unfortunately not.
I need a `msg_bus_entry_t` for each interface.

I could also use VLA instead :fearful:

Reply:
> Unfortunately not.
> I need a `msg_bus_entry_t` for each interface.

But we did so much work to not have this bound by a maximum size and it led to problems in the past already :-/. @jia200x what do you think? All in all, I would prefer it to be default 1 then, since that is what most our nodes usually have.

> I could also use VLA instead fearful

https://en.wikipedia.org/wiki/Vla?

Reply:
> > I could also use VLA instead fearful
> 
> https://en.wikipedia.org/wiki/Vla?

yummy, but I meant 

https://en.wikipedia.org/wiki/Variable-length_array#C99

They should generally be avoided because of their unpredictable nature and no way to check if stack space was exhausted.
However, this might actually be a valid use case: `msg_bus_entry_t` is only 12 byte and the number of netifs is also small, so it's unlikely that this will overflow the stack.

Reply:
I'm fine with it, since `gnrc_netif_numof` will always resolve to a relatively small number.

> However, this might actually be a valid use case: msg_bus_entry_t is only 12 byte and the number of netifs is also small, so it's unlikely that this will overflow the stack.

In fact, a big-enough compile time value might also overflow the stack. So let's go with it